main/windows/main.js

```diff
@@ -102,6 +103,26 @@ function setAspectRatio (aspectRatio) {
   main.win.setAspectRatio(aspectRatio)
 }
 
+function setContentSize (size) {
+  // Do nothing in fullscreen
+  if (!main.win || main.win.isFullScreen()) {
+    log('setContentSize: not setting content size because we\'re in full screen')
+    return
+  }
+
+  log('setContentSize: setting content size to ' + JSON.stringify(size))
+  var pos = main.win.getPosition()
+  var contentSize = main.win.getContentSize()
+  main.win.setContentSize(size.width, size.height, true)
+
+  // Keep the window centered at the same position as before resize
+  var x = Math.round(pos[0] + (contentSize[0] - size.width) / 2)
+  var y = Math.round(pos[1] + (contentSize[1] - size.height) / 2)
+
+  log('setContentSize: position changed to ' + JSON.stringify({ x, y }))
+  main.win.setPosition(x, y, false)
```

Comment:
We need to call this function because `BrowserWindow.setContentSize` keeps windows origin at the same position. This however poses a question if subsequently calling two functions modifying window state doesn't disturb resize animation on OS X.

Please confirm.